### PR TITLE
Refactor Player class

### PR DESCRIPTION
**Pull request by RaphaelBonatti**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/entity/Inventory.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import java.util.TimerTask;
+
+import ch.epfl.sdp.item.Healthpack;
+import ch.epfl.sdp.item.Item;
+import ch.epfl.sdp.item.Scan;
+import ch.epfl.sdp.item.Shield;
+import ch.epfl.sdp.item.Shrinker;
+
+public class Inventory {
+    private Player player;
```
Comment (kangpl) [inline app/src/main/java/ch/epfl/sdp/entity/Inventory.java]: Is it necessary to have Player field?

File: app/src/main/java/ch/epfl/sdp/entity/Inventory.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    public Inventory(Player player) {
+        this.player = player;
+        this.items = new HashMap<>();
+    }
+
+    public void useItem(Item i) {
+        int numberOfInstances = items.get(i.getName());
+
+        if (numberOfInstances > 0) {
```
Comment (varathar53) [inline app/src/main/java/ch/epfl/sdp/entity/Inventory.java]: enum or polymo?

File: app/src/main/java/ch/epfl/sdp/entity/Inventory.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    private void useHealthPack(Healthpack healthpack) {
+        player.healthPoints = player.healthPoints + healthpack.getHealthPackAmount();
+        if (player.healthPoints > Player.MAX_HEALTH) {
+            player.healthPoints = Player.MAX_HEALTH;
+        }
+    }
+
+    public void addItem(String itemName) {
+        int n = items.get(itemName);
+        n = n + 1;
```
Comment (varathar53) [inline app/src/main/java/ch/epfl/sdp/entity/Inventory.java]: n += n

File: app/src/main/java/ch/epfl/sdp/entity/Inventory.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        items.put(itemName, n);
+    }
+
+    public void removeItem(String itemName) {
+        int n = items.get(itemName);
+        if (n > 0) {
+            n = n - 1;
```
Comment (varathar53) [inline app/src/main/java/ch/epfl/sdp/entity/Inventory.java]: n -=n

File: app/src/main/java/ch/epfl/sdp/logic/RandomGenerator.java [hunk 5]
```diff
@@ -113,28 +111,28 @@ public Scan randomScan() {
...
-         Player p5 = new Player(g5.longitude(), g5.latitude(), rand.nextDouble()+50, randomString(10), randomEmail());
+         Player p5 = new Player(g5.getLongitude(), g5.getLatitude(), rand.nextDouble()+50, randomString(10), randomEmail());
 
-         List<Player> players = new ArrayList<>();
-         players.add(p1);
-         players.add(p2);
-         players.add(p3);
-         players.add(p4);
-         players.add(p5);
+         PlayerManager playerManager = new PlayerManager();
```
Comment (varathar53) [inline app/src/main/java/ch/epfl/sdp/logic/RandomGenerator.java]: Good Idea with the PlayerManager

**Pull request merged**